scripts/generate-sitemap.js

```diff
@@ -13,63 +13,99 @@ async function readSchemaJson(filePath) {
   }
 }
 
-async function processDirectory(dirPath, relativePath = "") {
-  const entries = await fs.readdir(dirPath, { withFileTypes: true });
-  let paths = [];
-
-  for (const entry of entries) {
-    const entryPath = path.join(dirPath, entry.name);
-
-    if (entry.name === "index.json") {
-      continue;
+async function processFile(fullPath, relativePath, name) {
+  const schemaPath = _path.join(fullPath, name);
+  const schemaData = await readSchemaJson(schemaPath);
+  if (!schemaData) {
+    return null;
+  }
+  const pageName = name.split(".")[0];
+  const permalink = _path.join(relativePath, pageName);
+  const siteMapEntry = {
+    permalink,
+    title:
+      schemaData.page.title ||
+      pageName.charAt(0).toUpperCase() + pageName.slice(1),
+  };
+  // check if file is actually an index page for a directory
+  try {
+    const directoryPath = _path.join(fullPath, name.split(".")[0]);
+    const stats = await fs.stat(directoryPath);
+    if (stats.isDirectory()) {
+      return {
+        ...siteMapEntry,
+        children: await processDirectory(directoryPath, permalink),
+      };
     }
+    return siteMapEntry;
+  } catch (error) {
+    return siteMapEntry;
+  }
+}
 
-    if (entry.isDirectory()) {
-      const subRelativePath = path.join(relativePath, entry.name);
-
-      // Check if index.json exists
-      const indexFilePath = path.join(entryPath, "index.json");
-      const indexSchemaData = await readSchemaJson(indexFilePath);
+async function processDirectory(fullPath, relativePath) {
+  const entries = await fs.readdir(fullPath, { withFileTypes: true });
+  let children = [];
 
-      if (indexSchemaData) {
-        const permalink = `/${subRelativePath}`;
-        paths.push({
-          permalink,
-          title: indexSchemaData.page.title || entry.name,
-          paths: await processDirectory(entryPath, subRelativePath),
-        });
-      } else {
-        paths.push({
-          paths: await processDirectory(entryPath, subRelativePath),
-        });
+  // Check if _pages.json exists
+  const pageOrderFilePath = _path.join(fullPath, "_pages.json");
+  const pageOrderData = await readSchemaJson(pageOrderFilePath);
+  if (pageOrderData) {
+    const childPages = pageOrderData["pages"];
+    for (const child of childPages) {
+      const childEntry = await processFile(
+        fullPath,
+        relativePath,
+        child + ".json"
+      );
+      if (childEntry) {
+        children.push(childEntry);
       }
     }
+  } else {
+    // If _pages.json does not exist, process files in the directory in arbitrary order
+    const fileEntries = entries.filter((entry) => entry.isFile());
```

Comment:
issue: discussed offline, the existence of the index page for a folder (e.g. `/about.json` for the `/about` folder) is a weak guarantee provided by the CMS. We should add a sane default page in case that page doesn't exist, so we don't accidentally unpublish all the pages within the folder.

Reply:
[0684b34](https://github.com/isomerpages/isomer-next-build/pull/1/commits/0684b348b3191e14bbf9be17d967b71b7257ad31)

starting to get a bit jank 💀